netutils/interface.py

```diff
@@ -66,6 +147,68 @@ def canonical_interface_name(interface, addl_name_map=None, verify=False):
     return interface
 
 
+def canonical_interface_name_list(interfaces, addl_name_map=None, verify=False, order=None, reverse=None):
+    """Function to return a list of interface's canonical name (fully expanded name).
+
+    Use of explicit matches used to indicate a clear understanding on any potential
+    match. Regex and other looser matching methods were not implmented to avoid false
+    positive matches. As an example, it would make sense to do "[P|p][O|o]" which would
+    incorrectly match PO = POS and Po = Port-channel, leading to a false positive, not
+    easily troubleshot, found, or known.
+
+    Args:
+        interfaces (list): List of interfaces you are attempting to expand.
+        addl_name_map (dict, optional): A dict containing key/value pairs that updates the base mapping. Used if an OS has specific differences. e.g. {"Po": "PortChannel"} vs {"Po": "Port-Channel"}. Defaults to None.
+        verify (bool, optional): Whether or not to verify the interface matches a known interface standard. Defaults to False.
+        order (str, optional): Determines what order the list of interfaces should be returned in. Defaults to None.
+        reverse (bool, optional): Specify if the order of the list should be reversed when setting an order. Defaults to None.
+
+    Returns:
+        list: List of the interfaces in their long form.
+
+    Raises:
+        ValueError: Raised if any interface name in list cannot be converted to its long form and verify parameter is set to true.
+
+    Example:
+        >>> from net_utility.interface import canonical_interface_name_list
+        >>> canonical_interface_name_list(["Gi1/0/1", "Gi1/0/2", "Eth1"])
+        ['GigabitEthernet1/0/1', 'GigabitEthernet1/0/2', 'Ethernet1']
+        >>> canonical_interface_name_list(["Gi1/0/1", "Po40", "Lo10"])
+        ['GigabitEthernet1/0/1', 'Port-channel40', 'Loopback10']
+        >>>
+    """
+    name_map = {}
+    name_map.update(BASE_INTERFACES)
+    canonical_interface_list = []
+    no_match_list = []
+
+    if reverse and not order:
+        raise ValueError("Order must be set to use reverse.")
+
+    if order and order not in INTERFACE_LIST_ORDERING_OPTIONS.keys():
+        raise ValueError(f"{order} is not one of the supported orderings")
+
+    if isinstance(addl_name_map, dict):
+        name_map.update(addl_name_map)
+
+    for interface in interfaces:
+        interface_type, interface_number = split_interface(interface)
```

Comment:
I was expecting to see a call to `canonical_interface_name` here, any reason to not leverage the existing function?

Reply:
Good point. Cleaned it up a bit.